backend/controller/timeline/events_async.go

```diff
@@ -0,0 +1,76 @@
+package timeline
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/alecthomas/types/optional"
+
+	ftlencryption "github.com/TBD54566975/ftl/backend/controller/encryption/api"
+	"github.com/TBD54566975/ftl/backend/controller/timeline/internal/sql"
+	"github.com/TBD54566975/ftl/backend/libdal"
+	"github.com/TBD54566975/ftl/internal/model"
+	"github.com/TBD54566975/ftl/internal/schema"
+)
+
+type AsyncExecuteEvent struct {
+	ID       int64
+	Duration time.Duration
+	AsyncExecute
+}
+
+func (e *AsyncExecuteEvent) GetID() int64 { return e.ID }
+func (e *AsyncExecuteEvent) event()       {}
+
+type AsyncExecute struct {
```

Comment:
We'll want the async type here somewhere no?

Reply:
Yup, added